src/decompiler.cc

```diff
@@ -418,20 +411,37 @@ struct Decompiler {
         }
       }
       case ExprType::Block: {
-        return Block(args[0], cast<BlockExpr>(n.e)->block, LabelType::Block, "block");
+        auto& val = args[0];
+        val.v.push_back(
+              cat("label ", cast<BlockExpr>(n.e)->block.label, ":"));
+        if (parent && parent->ntype != NodeType::Statements
```

Comment:
This isn't quite clear. This is the code that's responsible for wrapping `{}` around blocks? So long as we have a parent that isn't a statement, block, loop, or some ifs?

This is for blocks used as expressions, right? It would be good to see a test of this to see what it's supposed to do.

Reply:
Ok, added comment and test case.